### PR TITLE
Add on_exception hook to documentation

### PR DESCRIPTION
**Pull request by daniellepintz**
The `on_exception` hook was recently added in #9183 but I forgot to add it to the documentation. 
In this PR I also make some very minor fixes.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/common/lightning_module.rst [hunk 1]
```diff
@@ -379,7 +379,7 @@ If you need to do something with all the outputs of each `validation_step`, over
 
 Validating with DataParallel
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-When training using a `accelerator` that splits data from each batch across GPUs, sometimes you might
+When training using an `accelerator` that splits data from each batch across GPUs, sometimes you might
```
Comment (awaelchli) [inline docs/source/common/lightning_module.rst:382]: while we are at it here, can we change the word "master" to "main" just below this line, thanks.

File: docs/source/common/lightning_module.rst [hunk 2]
```diff
@@ -379,7 +379,7 @@ If you need to do something with all the outputs of each `validation_step`, over
...
 need to aggregate them on the master GPU for processing (dp, or ddp2).
```
Comment (awaelchli) [inline docs/source/common/lightning_module.rst:383]:
```suggestion
need to aggregate them on the main GPU for processing (dp, or ddp2).
```

File: docs/source/common/lightning_module.rst [hunk 3]
```diff
@@ -267,7 +267,7 @@ The matching pseudocode is:
 
 Training with DataParallel
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
-When training using a `accelerator` that splits data from each batch across GPUs, sometimes you might
+When training using an `accelerator` that splits data from each batch across GPUs, sometimes you might
 need to aggregate them on the master GPU for processing (dp, or ddp2).
```
Comment (awaelchli) [inline docs/source/common/lightning_module.rst:271]:
```suggestion
need to aggregate them on the main GPU for processing (dp, or ddp2).
```

**Review by awaelchli — APPROVED** [review]

File: docs/source/extensions/callbacks.rst [hunk 4]
```diff
@@ -395,6 +395,12 @@ on_keyboard_interrupt
 .. automethod:: pytorch_lightning.callbacks.Callback.on_keyboard_interrupt
     :noindex:
 
+on_exception
+^^^^^^^^^^^^^^^^^^^^^
```
Comment (awaelchli) [inline docs/source/extensions/callbacks.rst:399]:
```suggestion
on_exception
^^^^^^^^^^^^
```
this must match the title length

**Comment by daniellepintz** [comment]
Can I get one more review here please?

**Review by rohitgr7 — APPROVED** [review]

**Pull request merged**